### PR TITLE
Add a new Translate Component

### PR DESCRIPTION
**Pull request by jisqyv**
The new translate component, intended to replace YandexTranslate,
routes translations through a proxy server run by MIT. This server in
turn routes requests to a commercial translation service, currently
Amazon's service.

Communication is via protocol buffers for efficiency and ease of
upgrade. Instead of a single API key, each MIT App Inventor user has
their own API key for the MIT service. So if an API key is
compromised, we can block just that one key.

Change-Id: I5dbce2cfbf4071fb655d28fe64de73bd360e0a8b

### CODE REVIEW

**Comment by jisqyv** [comment]
Note: There is some duplication of code between the Translate Component and the CloudDB component. This has to do with managing trusted certificates, which is needed in order to deal with the expiration of the DST X3 root back last September (which affects older devices). The code here and the code in the CloudDB component can be merged at some point. I didn't do it here because I didn't want to delay getting this done. The extra time wouldn't be needed for the coding, but for the extensive testing I would want to do on the CloudDB component.

**Comment by jisqyv** [comment]
Renamed to Translator (and other fixes)

File: appinventor/appengine/src/com/google/appinventor/server/tokenauth/tr.proto
Comment (ewpatton) [inline appinventor/appengine/src/com/google/appinventor/server/tokenauth/tr.proto:44]: Minor thing, but it might be worth adding a newline at the end here.

File: appinventor/components/src/com/google/appinventor/components/runtime/translate/tr.proto
Comment (ewpatton) [inline appinventor/components/src/com/google/appinventor/components/runtime/translate/tr.proto:42]: Consider including a newline at the end of the file.

File: appinventor/appengine/src/com/google/appinventor/client/editor/simple/components/MockTranslator.java
Comment (ewpatton) [inline appinventor/appengine/src/com/google/appinventor/client/editor/simple/components/MockTranslator.java:22]: This seems like a weird linebreak.

File: appinventor/appengine/src/com/google/appinventor/client/editor/simple/components/MockTranslator.java
Comment (ewpatton) [inline appinventor/appengine/src/com/google/appinventor/client/editor/simple/components/MockTranslator.java:77]: Remove unused code.

File: appinventor/appengine/src/com/google/appinventor/client/editor/simple/components/MockTranslator.java
Comment (ewpatton) [inline appinventor/appengine/src/com/google/appinventor/client/editor/simple/components/MockTranslator.java:69]: Isn't it the case that the default behavior of isPropertyForYail() to return true? If so this is redundant.

File: appinventor/components/src/com/google/appinventor/components/runtime/Translator.java
Comment (ewpatton) [inline appinventor/components/src/com/google/appinventor/components/runtime/Translator.java:221]: For CloudDB we don't include our CA in the public sources. Was this intentional?

File: appinventor/components/src/com/google/appinventor/components/runtime/Translator.java
Comment (ewpatton) [inline appinventor/components/src/com/google/appinventor/components/runtime/Translator.java:231]: For consistency with field naming scheme it should be `sslSockFactory`.

**Review by ewpatton — CHANGES_REQUESTED** [review]

File: appinventor/appengine/src/com/google/appinventor/client/editor/simple/components/MockTranslator.java
Comment (jisqyv) [reply appinventor/appengine/src/com/google/appinventor/client/editor/simple/components/MockTranslator.java:69]: No, isPropertyForYail returns false for PROPERTY_NAME_NAME (in MockComponent.java)

File: appinventor/components/src/com/google/appinventor/components/runtime/Translator.java
Comment (jisqyv) [reply appinventor/components/src/com/google/appinventor/components/runtime/Translator.java:221]: Yes, this is intentional. When I refactor this code (combine the common code fro CloudDB and Translator) I'll leave this in the open source, as it doesn't hurt and the fewer changes we make in the branding, the better.

File: appinventor/components/src/com/google/appinventor/components/runtime/Translator.java
Comment (jisqyv) [reply appinventor/components/src/com/google/appinventor/components/runtime/Translator.java:231]: Agree. I'll change it

File: appinventor/appengine/src/com/google/appinventor/client/editor/simple/components/MockTranslator.java
Comment (jisqyv) [reply appinventor/appengine/src/com/google/appinventor/client/editor/simple/components/MockTranslator.java:22]: Yep, but intentional. I guess I should change "based" to "Based" so it is clear

**Comment by jisqyv** [comment]
Converted to Draft because I'm going to add a feature to support multiple key ids (Most of the work is in the proxy server) the only change required here is to label the keyid used. Commit coming soon.

**Comment by jisqyv** [comment]
@halatmit This PR is ready for review (again). I have made the change to support multiple keys, the change is already deployed on the proxy server. I have tested it and all works as expected. Please do not merge this change, just approve it with git (or just say LGTM). I'll squash it down to one commit, digitally sign it and merge it once I have the go ahead. Thanks.

File: appinventor/appengine/src/com/google/appinventor/server/tokenauth/tr.proto
Comment (halatmit) [inline appinventor/appengine/src/com/google/appinventor/server/tokenauth/tr.proto:25]: I don't see anything wrong here, but I don't understand the reasons for the values of version and generation (or what these really are).  Might be good to add a comment.

**Review by halatmit — COMMENTED** [review]
LGTM, except for suggestion to add a comment in tr.proto

**Review by ewpatton — APPROVED** [review]
LGTM.

**Comment by jisqyv** [comment]
I've squashed and rebased this change. I'll merge it after Jenkins gives it the thumbs up

**Pull request merged**
